### DISCUSSION TITLE
How do you add a new/custom/personal lang to emacs?

### DISCUSSION THREAD

**Discussion by laduke** [emacs-tree-sitter/elisp-tree-sitter#222] [Q&A]
Hello, 
emacs-tree-sitter is great. I'm using it for a few languages. 

I made a small grammer with tree-sitter {generate,parse,highlight} and it's working well enough. I'm not sure how to get it into emacs now. It would not be worth putting somewhere public like melpa or the tree-sitter-langs repo.

**Comment by jcs090218** [comment]
It's still great to contribute to `tree-sitter-langs` since you can use built-in `treesit.el` with [treesit-langs](https://github.com/emacs-tree-sitter/treesit-langs).
